### ISSUE TITLE
The application was unable to start correctly (0xc000007b)

### ISSUE THREAD

**Issue by ayzick** [milkdrop2077/MilkDrop3#66]
I just Recently Upgraded from a Windows 10 to Windows 11 and Now I'm getting this error every time I start the application, Now what am I doing wrong?

**Comment by milkdrop2077** [comment]
install the visual c++ redistributable: https://aka.ms/vs/17/release/vc_redist.x86.exe
and DirectX End-User Runtimes (June 2010): https://download.microsoft.com/download/8/4/A/84A35BF1-DAFE-4AE8-82AF-AD2AE20B6B14/directx_Jun2010_redist.exe

this should fix your problem

**Comment by ayzick** [comment]
I installed them both and it's still not opening

**Comment by milkdrop2077** [comment]
try this one maybe : https://www.majorgeeks.com/files/details/visual_c_redistributable_runtimes_aio_repack.html

**Comment by ayzick** [comment]
Still not working

**Comment by milkdrop2077** [comment]
Do a clean install of windows 11. upgrading creates problems...

**Comment by Dragodraki** [comment]
Cause I am working in IT for several years I faced this issue a few times and (hopefully) can help you with some (less known) solutions. Your case seems to fit scenario B.

A: Usually this error code appears if your PC shows BSOD / won't boot correctly (what means, an inlace upgrade is needed) - possible since Windows Vista. 
![grafik](https://github.com/milkdrop2077/MilkDrop3/assets/76787321/eec5f7e2-14cd-4a8b-85e7-074564aeea66)
The reason could vary, so better not waste you time trying to find out why exactly. If your PC don't boot up at all and you are not willing to do a clean install, you can try to install Windows to another physical drive first (or backup your damaged to another first and use your original as formatted 'new' one with 1:1 live image) and then copy/overwrite only the big data partition (not the little SYSTEM partition) from your backup drive to the with the fresh installed windows (with a live-image tool like MiniTool Partition Wizard PE).


B: In rare cases where it appears trying to run an single app (your PC boot up without errors and its NOT about an BSOD):
https://www.diskgenius.com/how-to/img/fix-error-code-0xc000007b-01.png
As far as I am aware 0xc000007b indicates an incorrect usage/mix of 32-bit and 64-bit where they shouldn't. We can be even more specific here: Your 'defect app' is not defect: The EXE binary is probably 32-Bit (x86) but at least one of its DLL files supports explicit 64-Bit (x64). You have to replace all 64-Bit DLLs with 32-Bit DLLs urgently and the app should run fine. Its not a problem with your OS and if its not a Microsoft app an upgrade or clean install wouln't change anything (if not scenario A).

**Comment by joakimlemb** [comment]
Just had the same issue on a clean Windows 11 install.
Milkdrop was looking for "D3DX9_43.dll" in the "c:\Windows\SysWOW64" folder.

Tried reinstalling DirectX multiple times without success.
Found a workaround by copying "D3DX9_43.dll" to the same folder as the Milkdrop3.exe binary from a different machine.

The software that helped find out I was missing that .dll file was: https://github.com/lucasg/Dependencies

**Comment by Dragodraki** [comment]
@joakimlemb
Yes, tools like "Dependendies" or "CFF Explorer" will show the required library files (.dll). However not every DLL is what I called portable like DirectX. 

Having that problem with Windows own files like advapi32.dll, ntdll.dll, kernel32.dll would definitely not work with downloading and copying to into game folder would definitely NOT work (registry association and mixing up with another dll files from OS are the cause). Otherwise every app would be compatible with every Windows system.

Here is a list of standard programs that support DLL-version cross-loading:
- C++ Redistributable
- Java
- DirectPlay
- QuickTime
- Adobe Shockwave Flash
- DirectX
- Steam
- Powershell
- OpenAL
- Browser Plugins

Here is a list of standard programs that does not (or only partly) support DLL-version cross-loading:
- Microsoft Windows
- Every app written as UWP
- NET Framework
- Nvidia PhysX

Thats the reason for why some apps cannot be build as portable standalone.
DLL-Troubling applications/games that based on standard programs from the unsupported list unfortunately can rarely being fixed by patches, except from the original developer.

**Comment by psydemekum** [comment]
Under Windows 10, using the file from the sysvol64 directory resolved the issue. However, after upgrading to Windows 11, the error persisted—until I replaced the file with the d3dx9_43.dll from this "ancient" archive: https://www.mediafire.com/file/zloviu3elfzbpty/Windows_Error_0xc000007b_H%2526H_Tutoriais.rar/file

**Comment by thexfiles123** [comment]
Dug up my github account to post here, thanks guys, I installed a billion c++ redists and directx files and nothing helped, until I copied d3dx9_43.dll from my system32 folder to the milkdrop folder, and suddenly it started working fine, I hope this is fixed with the new version though...

**Comment by turbokuk** [comment]
> Under Windows 10, using the file from the sysvol64 directory resolved the issue. However, after upgrading to Windows 11, the error persisted—until I replaced the file with the d3dx9_43.dll from this "ancient" archive: https://www.mediafire.com/file/zloviu3elfzbpty/Windows_Error_0xc000007b_H%2526H_Tutoriais.rar/file

Thanx a BIG bunch! It worked like a charm <3
